public/app/features/alerting/unified/components/rules/central-state-history/CentralHistoryRuntimeDataSource.ts

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useMemo } from 'react';
+
+import { DataQuery, DataQueryRequest, DataQueryResponse, TestDataSourceResponse } from '@grafana/data';
+import { RuntimeDataSource, sceneUtils } from '@grafana/scenes';
+import { getTimeSrv } from 'app/features/dashboard/services/TimeSrv';
+import { dispatch } from 'app/store/store';
+
+import { stateHistoryApi } from '../../../api/stateHistoryApi';
+import { DataSourceInformation } from '../../../home/Insights';
+
+import { LIMIT_EVENTS } from './EventListSceneObject';
+import { historyResultToDataFrame } from './utils';
+
+const historyDataSourceUid = '__history_api_ds_uid__';
+const historyDataSourcePluginId = '__history_api_ds_pluginId__';
+
+export const alertStateHistoryDatasource: DataSourceInformation = {
+  type: historyDataSourcePluginId,
+  uid: historyDataSourceUid,
+  settings: undefined,
+};
+
+export function useRegisterHistoryRuntimeDataSource() {
+  // we need to memoize the datasource so it is not registered multiple times for each render
+  const ds = useMemo(() => new HistoryAPIDatasource(historyDataSourceUid, historyDataSourcePluginId), []);
+  useEffect(() => {
+    sceneUtils.registerRuntimeDataSource({ dataSource: ds });
+  }, [ds]);
+}
+
+/**
+ * This class is a runtime datasource that fetches the events from the history api.
+ * The events are grouped by alert instance and then converted to a DataFrame list.
+ * The DataFrame list is then grouped by time.
+ * This allows us to filter the events by labels.
+ * The result is a timeseries panel that shows the events for the selected time range and filtered by labels.
+ */
+class HistoryAPIDatasource extends RuntimeDataSource {
+  constructor(pluginId: string, uid: string) {
+    super(uid, pluginId);
+  }
+
+  async query(request: DataQueryRequest<DataQuery>): Promise<DataQueryResponse> {
+    const from = request.range.from.unix();
+    const to = request.range.to.unix();
+
+    return {
+      data: historyResultToDataFrame(await getHistory(from, to)),
+    };
+  }
+
+  testDatasource(): Promise<TestDataSourceResponse> {
```

Comment:
I was not sure what to put in here, but I saw that this is the common code in several places

Reply:
Hmm I guess the test implementation would probably be something like calling the history API via RTKQ and unwrapping the response to check? But this seems fine if it's what others are also doing :+1:

Reply:
I believe we should request to a health endpoint, but I think we don't have one for it